README.md

```diff
@@ -1,42 +1,106 @@
+# Creating your own codec
+
+You can create a codec for your protocol by implementing the following interface - [com.exactpro.sf.externalapi.codec.IExternalCodec](https://github.com/exactpro/sailfish-core/blob/master/BackEnd/Core/sailfish-core/src/main/kotlin/com/exactpro/sf/externalapi/codec/IExternalCodec.kt).
+Also, you will need to implement the [com.exactpro.sf.externalapi.codec.IExternalCodecFactory](https://github.com/exactpro/sailfish-core/blob/master/BackEnd/Core/sailfish-core/src/main/kotlin/com/exactpro/sf/externalapi/codec/IExternalCodecFactory.kt) interface.
+
+The core part of the "Codec" component uses [ServiceLoader](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ServiceLoader.html) to load all the factory interface implementations.
+To provide the ServiceLoader with the knowledge about your factory implementation the JAR file should contain a provider-configuration file named:
+
+**META-INF/services/com.exactpro.sf.externalapi.codec.IExternalCodecFactory**
+
+with the content equals to the fully-qualified class name of your factory implementation.
+
+_If you have several implementations of that interface their fully-qualified names should be written in that file each one on the new line._
+
 
 # Configuration
 
-Codec has got four types of connection: stream and general for encode and decode functions.
+Codec has four types of connection: stream and general for encode and decode functions.
 
-* stream encode / decode connections are used for work 24 / 7
-* general encode / decode connections are used for requests on demand
+* stream encode / decode connections works 24 / 7
+* general encode / decode connections works on demand
 
-Messages in stream and general connections are never mixed. 
+Codec never mixes messages from the _stream_ and the _general_ connections. 
 
 Decoding can work in two different modes:
-+ CUMULATIVE (default) - all raw messages in batch will be joined together and decoded. After decoding, content and count of the decoded messages will be compared with the original messages in the batch.
-+ SEQUENTIAL - each message in the batch will be decoded as separate message.
++ **CUMULATIVE** (default) - all raw messages in batch will be joined together and decoded. After decoding, content and count of the decoded messages will be compared with the original messages in the batch.
++ **SEQUENTIAL** - each message in the batch will be decoded as separate message.
 
 This setting can be overridden in a custom config for the application using `decodeProcessorType` parameter.
 
-## Requried pins
+## Bootstrap parameters
 
-Every type of connections is presented two subscribe and publish pins. 
-Configuration should include at least one at a time pins for every types (minimal number of them is 8)
+These parameters specify the codec to be used for the messages decoding/encoding and the mode which should be used.
+Their should be defined in the `custom-config` section of the component configuration.
```

Comment:
**They** should be defined in the `custom-config` section of the component configuration.